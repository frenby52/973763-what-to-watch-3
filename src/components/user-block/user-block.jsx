import React from 'react';
import PropTypes from "prop-types";
import {getAuthUser, isAuth} from "../../reducer/user/selectors.js";
import {connect} from "react-redux";
import {Link} from 'react-router-dom';

const BASE_URL = `https://htmlacademy-react-3.appspot.com/`;

export const UserBlock = (props) => {
  const {isAuthed, authUserData} = props;

  return (
    <div className="user-block">
      {isAuthed
        ? <div className="user-block__avatar">
          <Link to="/mylist">
            <img src={`${BASE_URL}${authUserData.avatarUrl}`} alt={authUserData.name} width="63" height="63"/>
          </Link>
        </div>
        : <Link to='/login' className="user-block__link">Sign in</Link>}
    </div>
  );
};

UserBlock.propTypes = {
  isAuthed: PropTypes.bool.isRequired,
  authUserData: PropTypes.shape({
    id: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    avatarUrl: PropTypes.string.isRequired,
  }),
};

const mapStateToProps = (state) => ({
  isAuthed: isAuth(state),
  authUserData: getAuthUser(state),
});

export default connect(mapStateToProps)(UserBlock);
