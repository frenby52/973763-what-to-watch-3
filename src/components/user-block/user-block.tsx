import * as React from "react";
import {getAuthUser, isAuth} from "../../reducer/user/selectors";
import {connect} from "react-redux";
import {Link} from 'react-router-dom';

const BASE_URL = `https://htmlacademy-react-3.appspot.com/`;

type UserBlockProps = {
  authUserData: {
    id: number;
    email: string;
    name: string;
    avatarUrl: object;
  };
  isAuthed: boolean;
};

export const UserBlock: React.FunctionComponent<UserBlockProps> = (props: UserBlockProps) => {
  const {isAuthed, authUserData} = props;

  return (
    <div className="user-block">
      {isAuthed
        ? <div className="user-block__avatar">
          <Link to="/mylist">
            <img src={`${BASE_URL}${authUserData.avatarUrl}`} alt={authUserData.name} width="63" height="63"/>
          </Link>
        </div>
        : <Link to='/login' className="user-block__link">Sign in</Link>}
    </div>
  );
};

const mapStateToProps = (state) => ({
  isAuthed: isAuth(state),
  authUserData: getAuthUser(state),
});

export default connect(mapStateToProps)(UserBlock);
