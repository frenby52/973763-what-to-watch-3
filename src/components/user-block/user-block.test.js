import React from 'react';
import renderer from 'react-test-renderer';
import {UserBlock} from "./user-block.jsx";

const authUserData = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

jest.mock(`react-router-dom`, () => ({Link: `Link`}));

it(`UserBlock component should render correct`, () => {
  const tree = renderer
    .create(
        <UserBlock isAuthed={true} authUserData={authUserData}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
