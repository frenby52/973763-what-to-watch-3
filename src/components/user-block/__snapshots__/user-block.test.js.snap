// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock component should render correct 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <Link
      to="/mylist"
    >
      <img
        alt="Oliver.conner"
        height="63"
        src="https://htmlacademy-react-3.appspot.com/img/1.png"
        width="63"
      />
    </Link>
  </div>
</div>
`;
